xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/test/java/org/xwiki/attachment/internal/configuration/DefaultAttachmentConfigurationTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment.internal.configuration;
+
+import javax.inject.Named;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test of {@link DefaultAttachmentConfiguration}.
+ *
+ * @version $Id$
+ * @since 16.3.0RC1
+ */
+@ComponentTest
+class DefaultAttachmentConfigurationTest
+{
+    @InjectMockComponents
+    private DefaultAttachmentConfiguration attachmentConfiguration;
+
+    @MockComponent
+    @Named("xwikiproperties")
+    private ConfigurationSource xWikiPropertiesConfigurationSource;
+
+    private static final String ATTACHMENT_MIMETYPE_ENABLE_COMMENTS_PROPERTY = "attachment.upload.enableComments";
+
+    @ParameterizedTest
+    @CsvSource({"true", "false"})
```

Comment:
`@ValueSource(booleans = { true, false })` seems simpler to me.

Reply:
As a general rule I'm not very fond of parameterized tests for functional tests since func tests are supposed to be scenarios and there's often some running duplications with parameterized tests (to reset the initial conditions). I'm not saying it's bad here, I haven't checked. But you need to keep in mind that we want the minimal number of functional tests needed (if some tests can be written as unit tests or page tests, it's better), and to be very conscious about the time it takes to execute them (our build is already very long and running on all supported configurations make it even slower). So when you make a parameterized test with N options, for example you add N times the number of environments.

Reply:
This is not a functional test, this is a very simple unit test of a single component.

Reply:
oops my bad then, that'll teach me to speak without re-checking first ;) (I had checked a test earlier and it was a functional tests, but I didn't recheck). Sorry guys (but what I mentioned is still valid and there are plenty of existing functional tests that take too much time)